src/components/FindMovie/FindMovie.tsx

```
@@ -1,10 +1,73 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import classNames from 'classnames';
+
+export const FindMovie: React.FC = ({ onChangeMovies }) => {
+  const [query, setQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  function isMovieData(data): data is MovieData {
+    return (
+      typeof data === 'object' &&
+      data !== null &&
+      'Poster' in data &&
+      'Title' in data &&
+      'Plot' in data &&
+      'imdbID' in data
+    );
+  }
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    setLoading(true);
+
+    getMovie(query)
+      .then((movieData: MovieData | ResponseError) => {
+        if (isMovieData(movieData)) {
+          setMovie({
+            title: movieData.Title,
+            description: movieData.Plot,
+            imgUrl:
+              movieData.Poster.startsWith('http://') ||
+              movieData.Poster.startsWith('https://')
+                ? movieData.Poster
+                : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+            imdbUrl: `https://www.imdb.com/title/${movieData.imdbID}`,
+            imdbId: movieData.imdbID,
+          });
+        } else {
+          setErrorMessage(movieData.Error);
+        }
+      })
+      .catch()
```

An empty `.catch()` will suppress any errors that might occur within the `.then()` block, making them difficult to debug. While `getMovie` handles fetch errors, any runtime error inside the `.then()` logic will be silently ignored. It's better to at least log potential errors, for example, `.catch(console.error)`.